messaging/src/main/java/org/axonframework/messaging/FilteringMessageStream.java

```diff
@@ -35,6 +35,7 @@ class FilteringMessageStream<M extends Message<?>> implements MessageStream<M> {
 
     private final MessageStream<M> delegate;
     private final Predicate<Entry<M>> filter;
+    private Entry<M> lookAhead = null;
```

Comment:
My gut tells me this field should be called `peeked` instead of `lookAhead`. Simply because the method setting this is called `peek()`, not `lookAhead()` (although it is essentially the same).

If anything, a massive nit. I don't mind if you prefer it to be named as it is right now.
If you do decide to adjust it to `peeked`, than that change should be propagated to the other `MessageStream` implementations that maintain the upcoming value (or we call it `upcomingEntry`! :p).